misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/BallerinaStackFrame.java

```diff
@@ -128,13 +133,37 @@ private static String getStackFrameName(StackFrameProxyImpl stackFrame) {
                 return stackFrame.visibleVariableByName(STRAND_VAR_NAME) != null ? frameName :
                         FRAME_TYPE_START + FRAME_SEPARATOR + frameName;
             } else {
-                return stackFrame.location().method().name();
+                return getFilteredStackFrame(stackFrame);
             }
         } catch (Exception e) {
             return FRAME_TYPE_ANONYMOUS;
         }
     }
 
+    /**
+     * Derives filtered ballerina stack frame name from the given java stack frame instance.
+     *
+     * @param stackFrame JDI stack frame instance
+     * @return filtered ballerina stack frame name
+     */
+    private static String getFilteredStackFrame(StackFrameProxyImpl stackFrame) throws JdiProxyException {
+        Value value = stackFrame.getValue(stackFrame.visibleVariables().get(0));
+        String stackFrameName = stackFrame.location().method().name();
+        if (isService(value)) {
+            if (stackFrameName.equals(METHOD_INIT)) {
+                return BVariableType.SERVICE.getString();
+            } else {
+                // Here stackFrameName format will be `$resourceAccessor$resourceName$otherResourceParts`
+                String[] serviceResources = stackFrameName.split("\\$");
+                if (serviceResources[1].equals(ACCESSOR_DEFAULT)) {
```

Comment:
Add a check to verify the size before accessing `serviceResources` by index.

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/31279/commits/8b8ba6a04de50d8d9a98f0622372775b3c2b9262

Reply:
Still not fixed

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/31279/commits/19e70a018836e09ced64ed7c8ce5c527328237f4